### ISSUE TITLE
Add intellisense support

### ISSUE THREAD

**Issue by yorek** [rickbergfalk/sqlpad#523]
Intellisense would be a great addition. Happy to help if needed

**Comment by rickbergfalk** [comment]
Hi @yorek - By Intellisense do you mean Intellisense proper by Microsoft or as in some sort of autocomplete?

There is a very naive implementation that will suggest tables and columns... though its kinda clunky and tries to be smart. Tables need to be added to the `FROM` clause first, and table aliases aren't supported.

**Comment by yorek** [comment]
Hi @rickbergfalk, I mean that I (me, not Microsoft) would love to help :)
I didn't notice that after the table has been specified, intellisense will provide more options. I'll check that out

**Comment by rickbergfalk** [comment]
It definitely could use a more proper implementation. The existing implementation was something quickly put together a couple years ago

I've also been wanting to transition the editor to [monaco](https://microsoft.github.io/monaco-editor/) if that is of interest.

**Comment by rickbergfalk** [comment]
I came to the realization that there might have been a bug around this -- autocomplete wouldn't happen if the sidebar was closed on initial load. 

Another potential issue here is auto complete is maybe trying to be too smart. It won't suggest any column names until something is in the FROM clause, and even then column names will only be suggested if you're in a clause where it makes sense.

I think there's lots of room for improvement though. 

@yorek I'm going to close this issue, but if you or someone else want to take on enhancing this we can re-open this issue or open a new one.
